Neos.Cache/Classes/Backend/RedisBackend.php

```diff
@@ -456,22 +456,20 @@ public function setRedis(\Redis $redis = null)
     }
 
     /**
-     * TODO: No return type declaration for now, as it needs to return false as well.
-     * @param string $value
-     * @return mixed
+     * @param string|bool $value
+     * @return string|bool
      */
-    private function uncompress(string $value): string
+    private function uncompress($value)
     {
-        if (empty($value)) {
+        if ($value === false || empty($value)) {
```

Comment:
`empty(false) === true` therefore this addtional check is unnecessary.

Reply:
Yes, but it clearly transports the intention and does away with implicit type handling. Thinking about it, this should probably even be `if ($value === false || $value === '') {} elseif (!is_string($value) { throw new \InvalidArgumentException(); }`